base_decorator.rb

```diff
@@ -0,0 +1,7 @@
+require './nameable'
+class BaseDecorator < Nameable
+  def correct_name
+    super()
+    @nameable.correct_name
+  end
+end

```

Comment:
- [ ] Kindly create the getters and setters for @nameable `attr_accessor :nameable`
- Instance variables are the only ones that should be able to be modified that's why they are defined as attr_accessors (here both Getters and Setters are defined).